src/Pelagos/Bundle/AppBundle/Resources/public/js/datasetReview.js

```diff
@@ -656,8 +656,28 @@ $(document).ready(function(){
               }
         });
     });
+
+    checkColdStorageCheckBoxState();
+    //coldstorage checkbox - distribution info tab
+    $(".coldstorage-checkbox").change(function (){
+        checkColdStorageCheckBoxState();
+    });
 });
 
+function checkColdStorageCheckBoxState() {
+    if ($(".coldstorage-checkbox").is(":checked")) {
+        $(".row-coldstorage-filesize").show();
+        $(".row-coldstorage-sha256hash").show();
+        $("#datasetFileColdStorageArchiveSize").attr("required", "required");
+        $("#datasetFileColdStorageArchiveSha256Hash").attr("required", "required");
+    } else {
+        $(".row-coldstorage-filesize").hide();
+        $(".row-coldstorage-sha256hash").hide();
+        $("#datasetFileColdStorageArchiveSize").removeAttr("required");
```

Comment:
Clear out the value (val("")), or else if you had marked it previously and unmarking doesn't clear the field, It would still persist the value when you submit the form

Reply:
There is a use case where user loads the review that has existing values, if they accidentally toggle the switch, the values would be cleared (user may panic because they can't undo it).
If they leave the value, the form will know how to clear the value using on the PRE_SUBMIT event.

Reply:
Update: I add a dialog box to prevent user from disabling the fields and empty the values by accident

Reply:
👍